### PR TITLE
DOC: Clarify Documentation for scipy.special.gammaln

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -1225,12 +1225,24 @@ def add_newdoc(place, name, doc):
     """
     gammaln(z)
 
-    Logarithm of absolute value of gamma function
+    Performs a logarithmic transformation of the
+    values of the gamma function in one of two
+    ways, depending on the input `z`:
 
-    Defined as::
+    1) type(z) != <type 'complex'>
```
Comment (rkern) [inline scipy/special/add_newdocs.py]: I'd suggest wording this like so:

```
1) `z` is not complex:
...
```

Strictly speaking, `type(z)` is likely `ndarray`! It's the `dtype` that matters.

Comment (gfyoung) [inline scipy/special/add_newdocs.py]: Ah, fair enough.  It escaped my mind for some reason that ```gammaln``` can broadcast.

**Comment by gfyoung** [comment]
@rkern : Updated with a happy Travis.

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -1225,12 +1225,28 @@ def add_newdoc(place, name, doc):
...
+    values of the gamma function in one of two
+    ways, depending on the input `z`:
 
-    Defined as::
+    1) `z` is not complex (i.e. `z` is a purely
+       real number *or* it is array_like and
+       contains purely real elements)
+
+    The natural logarithm of the absolute value of
+    gamma(z) is performed. Thus, it is defined as:
```
Comment (nmayorov) [inline scipy/special/add_newdocs.py]: Maybe "is computed" (below too)?

Comment (gfyoung) [inline scipy/special/add_newdocs.py]: Sounds good.  Will change.

File: scipy/special/__init__.py [hunk 3]
```diff
@@ -280,7 +280,7 @@
    :toctree: generated/
 
    gamma        -- Gamma function.
-   gammaln      -- Log of the absolute value of the gamma function.
+   gammaln      -- Log transformation of the gamma function.
```
Comment (nmayorov) [inline scipy/special/__init__.py]: Did you use the word "transformation" to point out that it is not always log of the gamma function? Looks like a good solution.

Comment (gfyoung) [inline scipy/special/__init__.py]: Yes, indeed.

**Comment by nmayorov** [comment]
It is related to issue #5590. I personally think, that the description now is good.

@rgommers could you please take a look and maybe merge this.

**Comment by rgommers** [comment]
LGTM, merged. Thanks all.

**Pull request merged**